### PR TITLE
Fix the test for shuffle function

### PR DESCRIPTION
**Pull request by geraint0923**

### CODE REVIEW

**Comment by raghavsethi** [comment]
/s/Fix the test for shuffle function/Fix flaky array_shuffle test

File: presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java [hunk 1]
```diff
@@ -7067,19 +7067,20 @@ public void testTimeLiterals()
...
 
             // check if the result is a correct permutation
             assertEqualsIgnoreOrder(result, expected);
 
-            // check if there is any duplicated result
-            assertTrue(resultSet.add(result));
+            resultSet.add(result);
         }
+        // Make sure there are more than 8 distinct results
+        assertTrue(resultSet.size() > 8);
```
Comment (raghavsethi) [inline presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java]: remove this comment and call resultSet distinctResults

File: presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java [hunk 2]
```diff
@@ -7067,19 +7067,20 @@ public void testTimeLiterals()
     public void testArrayShuffle()
             throws Exception
     {
-        List<Integer> expected = asList(2, 8, 7, 5, 4, 6, 3);
+        List<Integer> expected = asList(108, 37, 2, 8, 7, 5, 4, 6, 3, 1, 0, 9, 23, 10, 25);
```
Comment (raghavsethi) [inline presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java]: Make this 1, 2, 3... no reason for this to be confusing

File: presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java (hunk 1 above)
Comment (haozhun) [inline presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java]: What about `assertTrue(distinctResults.size() >= 8, "shuffle must produce at least 8 distinct results")`?

File: presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java [hunk 3]
```diff
@@ -7067,19 +7067,19 @@ public void testTimeLiterals()
...
+            List<Integer> result = (List<Integer>) computeActual(format("SELECT shuffle(ARRAY %s)", expected)).getOnlyValue();
 
             // check if the result is a correct permutation
             assertEqualsIgnoreOrder(result, expected);
 
-            // check if there is any duplicated result
-            assertTrue(resultSet.add(result));
+            distinctResults.add(result);
         }
+        assertTrue(distinctResults.size() >= 8, "shuffle must produce at least 8 distinct results");
```
Comment (raghavsethi) [inline presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java]: assertGreaterThan

File: presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java [hunk 4]
```diff
@@ -7067,19 +7067,19 @@ public void testTimeLiterals()
     public void testArrayShuffle()
             throws Exception
     {
-        List<Integer> expected = asList(2, 8, 7, 5, 4, 6, 3);
-        Set<List<Integer>> resultSet = new HashSet<>();
+        List<Integer> expected = asList(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18);
```
Comment (electrum) [inline presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java]:
```java
List<Integer> = IntStream.rangeClosed(1, 500).boxed().collect(toList());

File: presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java [hunk 5]
```diff
@@ -7067,19 +7067,19 @@ public void testTimeLiterals()
     public void testArrayShuffle()
```
Comment (electrum) [inline presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java]: This should be in `TestArrayOperators`, not in `AbstractTestQueries`

**Pull request merged**